sycl/source/detail/windows_pi.cpp

```diff
@@ -17,6 +17,16 @@ namespace sycl {
 namespace detail {
 namespace pi {
 
+// Performs an OS-level one-time initialization of PI.
+void initOsOnce() {
+  // Tells the system to not display the critical-error-handler message box.
+  // Instead, the system sends the error to the calling process.
+  // This is crucial for graceful handling of plugins that couldn't be
+  // loaded, e.g. due to missing native run-times.
+  //
+  (void)SetErrorMode(SEM_FAILCRITICALERRORS);
```

Comment:
I'm not sure we can do this since it modifies behavior of user application.
Could you please provide more information about the problems with handling plugins loading?

Reply:
Because of various reasons (e.g. missing native RT installation) and depending on system setting a modal message box is displayed to the user (and application freezes):
![image](https://user-images.githubusercontent.com/48694368/113729697-1a9d7480-96ac-11eb-81dc-bf41d5ebac8a.png)

Moreover this is the recommended usage: https://docs.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-seterrormode: 
> Best practice is that all applications call the process-wide SetErrorMode function with a parameter of SEM_FAILCRITICALERRORS at startup. This is to prevent error mode dialogs from hanging the application.

To be less intrusive we could change the setting only for loading plugins, and restore back after plugins initialization.
Would you prefer that ?

Reply:
I think so, there still be risk that it can affect a user if they have additional threads doing something non-SYCL related.
Can we have a toggle to disable setting this mode?

Reply:
On the other hand if this is going to address hangs in automated testing, probably, it's better to update tests to set this mode. What happens with these errors when SetErrorMode(SEM_FAILCRITICALERRORS) is set?
The doc says:
``` Instead, the system sends the error to the calling process. ```
But it is not quite clear which function gets the error.

Reply:
> Can we have a toggle to disable setting this mode?

I think we absolutely want to unconditionally suppress message boxes during plugins loading. If we choose to *only* affect plugins initialization (by restoring back users settings afterwards) then we don't need a toggle.

> But it is not quite clear which function gets the error.

I think it would be through https://docs.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-getlasterror but I didn't try that (and wouldn't want to do it as a part of this PR).

Reply:
Could you please clarify what happens then if there is an error like missing ze_loader.dll if `SetErrorMode(SEM_FAILCRITICALERRORS)` is done?

Reply:
the `LoadLibraryA` returns NULL, and the Level-Zero plugin is not loaded, ignored

Reply:
@romanovvlad , do you need more clarifications? can we merge it?